clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -389,4 +389,29 @@ public void testByteBufferSerializer() {
             assertArrayEquals(bytes, serializer.serialize(topic, directBuffer1));
         }
     }
+
+    @Test
+    public void testBooleanSerializer() {
+        byte[] testData = new byte[1];
+        testData[0] = (byte) 1;
+
+        Serde<Boolean> booleanSerde = Serdes.Boolean();
+        assertArrayEquals(testData, booleanSerde.serializer().serialize(topic, true));
+    }
+
+    @Test
+    public void testBooleanDeserializer() {
+        byte[] testData = new byte[1];
+        testData[0] = (byte) 1;
+
+        Serde<Boolean> booleanSerde = Serdes.Boolean();
+        assertEquals(true, booleanSerde.deserializer().deserialize(topic, testData));
+    }
+
+    @Test
+    public void booleanDeserializerShouldThrowOnNotNullValues() {
```

Comment:
perhaps rename this to `booleanDeserializerShouldThrowOnEmptyInput` ?

Reply:
Thank you for all suggestions. I've written a ParametrizedTest for Deserializer too, and in that case I've started to get warnings. The discussion about it is below.